store/SPEC_STANDARD.md

```diff
@@ -0,0 +1,110 @@
+## What is a store specification document?
+
+A store specification document is a design document describing a particular store feature such as a store interface or implementation that it is expected to be used (or already in use) by the Cosmos SDK. This also includes other store-related features such as caches and wrappers.
+
+## Sections
+
+An store specification document consists of, synopsis, overview and basic concepts, technical specification, history log, and copyright notice. All top-level sections are required. References should be included inline as links, or tabulated at the bottom of the section if necessary.  Included sub-sections should be listed in the order specified below. 
+
+### Synopsis
+
+The document should include a brief (~200 word) synopsis providing a high-level
+description of and rationale for the specification.
+
+### Overview and basic concepts
+
+This section should include a motivation sub-section and a definitions sub-section if required:
+
+- *Motivation* - A rationale for the existence of the proposed interface, implementation, or the proposed changes to an existing feature.
+- *Definitions* - A list of new terms or concepts utilised in the document or required to understand it.
+
+### System model and properties
+
+This section should include an assumptions sub-section if any, and the mandatory properties sub-section. Note that both sub-sections are tightly coupled: how to enforce a property will depend directly on the assumptions made.
+
+- *Assumptions* - A list of any assumptions made by the feature designer. This may for instance include how it is assumed that the feature will be used by upper layers in the stack.
+- *Properties* - The content of this sub-section depends on the type of specification. For implementations, the sub-section should include a list of properties that the implementation guarantees and list which properties it does not. The latter is as important as the former. In the case of a store interface, this sub-section should include a list of the desired properties that an implementation of the interface should guarantee. Ideally, along with each of the properties should go a text arguing why the property is required. Examples of properties are whether the feature is or must be thread-safe, atomicity-related properties or behaviour under failures.
+
+### Technical specification
+
+This is the main section of the document, and should contain protocol documentation, design rationale,
```

Comment:
Perhaps please list the requirements in bullet form so:
should contain:
* protocol documentation
* design rational
* required references
* technical details where appropriate